hydra-node/test/Hydra/Chain/Direct/TxSpec.hs

```diff
@@ -188,68 +208,78 @@ spec =
-                 in property
-                      ( all
-                          (`notElem` blueprintOutputs)
-                          commitOutputs
-                      )
-                      & counterexample ("blueprint outputs: " <> show blueprintOutputs)
-                      & counterexample ("commit outputs: " <> show commitOutputs)
-              , let blueprintSigs = blueprintBody ^. reqSignerHashesTxBodyL
-                    commitSigs = commitTxBody ^. reqSignerHashesTxBodyL
-                 in property (blueprintSigs `Set.isSubsetOf` commitSigs)
-                      & counterexample ("blueprint signatures: " <> show blueprintSigs)
-                      & counterexample ("commit signatures: " <> show commitSigs)
-              , let blueprintRefInputs = blueprintBody ^. referenceInputsTxBodyL
-                    commitRefInputs = commitTxBody ^. referenceInputsTxBodyL
-                 in property (blueprintRefInputs `Set.isSubsetOf` commitRefInputs)
-                      & counterexample ("blueprint reference inputs: " <> show blueprintRefInputs)
-                      & counterexample ("commit reference inputs: " <> show commitRefInputs)
-              ]
-
-getAuxMetadata :: AlonzoTxAuxData LedgerEra -> Map Word64 Metadatum
-getAuxMetadata (AlonzoTxAuxData metadata _ _) = metadata
+          forAllBlind genBlueprintTxWithUTxO $ \(lookupUTxO, blueprintTx) ->
+            counterexample ("Blueprint tx: " <> renderTxWithUTxO lookupUTxO blueprintTx) $ do
+              let createdTx =
+                    commitTx
+                      networkId
+                      scriptRegistry
+                      (mkHeadId Fixture.testPolicyId)
+                      ownParty
+                      CommitBlueprintTx{lookupUTxO, blueprintTx}
+                      (healthyInitialTxIn, toUTxOContext healthyInitialTxOut, verificationKeyHash ownVerificationKey)
+              counterexample ("\n\n\nCommit tx: " <> renderTxWithUTxO lookupUTxO createdTx) $ do
+                let blueprintBody = toLedgerTx blueprintTx ^. bodyTxL
+                let commitTxBody = toLedgerTx createdTx ^. bodyTxL
+                let spendableUTxO =
+                      UTxO.singleton (healthyInitialTxIn, toUTxOContext healthyInitialTxOut)
+                        <> lookupUTxO
+                        <> registryUTxO scriptRegistry
+
+                conjoin
+                  [ propTransactionEvaluates (blueprintTx, lookupUTxO)
+                      & counterexample "Blueprint transaction failed to evaluate"
+                  , propTransactionEvaluates (createdTx, spendableUTxO)
+                      & counterexample "Commit transaction failed to evaluate"
+                  , conjoin
+                      [ getAuxMetadata blueprintTx `propIsSubmapOf` getAuxMetadata createdTx
+                          & counterexample "Blueprint metadata incomplete"
+                      , propHasValidAuxData blueprintTx
```

Comment:
`propHasValidAuxData blueprintTx` seems redundant.

my reasoning: 
if getAuxMetadata blueprintTx `propIsSubmapOf` getAuxMetadata createdTx

and

propHasValidAuxData createdTx

does not imply propHasValidAuxData blueprintTx?

Reply:
No unfortunately not. The validation checks whether the auxiliary data hash and the metadata also line up. The submap check does not involve the auxiliary data.

Checking aux data hash is important as mlabs seems to still have problems with it.